manifests/config.pp

```diff
@@ -147,13 +147,18 @@
     }
   }
 
+  # $_legacy_pid is used to kill the old stunnel process set up from a previous
+  #   version of this module. It should be set to $pid, unless $pid is unset
+  #   and the system has systemd.
   $on_systemd = 'systemd' in $facts['init_systems']
   if ($pid =~ Undef and $on_systemd) {
     $_foreground = true
     $_pid        = $pid
+    $_legacy_pid = '/var/run/stunnel/stunnel.pid'
   } else {
     $_foreground = undef
     $_pid        = '/var/run/stunnel/stunnel.pid'
+    $_legacy_pid = '/var/run/stunnel/stunnel.pid'
```

Comment:
Just set this once outside the `if/else` block

Reply:
If the `$pid` is specified by the user, I don't want to override that... But this as it stands doesn't do that.